pkg/runtime/wfengine/workflow.go

```diff
@@ -184,20 +196,54 @@ func (wf *workflowActor) createWorkflowInstance(ctx context.Context, actorID str
 		}
 	}
 
-	// We block (re)creation of existing workflows unless they are in a completed state
-	// Or if they still have any pending activity result awaited.
-	if !created {
-		runtimeState := getRuntimeState(actorID, state)
-		if !runtimeState.IsCompleted() {
-			return fmt.Errorf("an active workflow with ID '%s' already exists", actorID)
-		}
-		if wf.activityResultAwaited.Load() {
-			return fmt.Errorf("a terminated workflow with ID '%s' is already awaiting an activity result", actorID)
+	// orchestration didn't exist and just create it.
+	if created {
+		return wf.createInstance(ctx, actorID, startEvent, state)
+	}
+
+	// orchestration already exists, apply reuse id policy
+	runtimeState := getRuntimeState(actorID, state)
+	runtimeStatus := runtimeState.RuntimeStatus()
+	targetStatusValues := backend.BuildStatusSet(reuseIDPolicy.GetOperationStatus())
```

Comment:
This goes back to my comment on the durabletask-go repo, but I don't think we need to expose a `BuildStatusSet` utility function for this simple purpose. I think a simple for-loop over `reuseIDPolicy.GetOperationStatus()` and comparing to `runtimeStatus` on each loop iteration is good enough since we only have a small number of possible status values. This might even be more performant than creating a map and doing hashing operations for the key lookup.